bin/account-ids-by-name.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```

Comment:
Gathering the account names and account IDs from filenames _feels_ brittle to me, but the only better idea I have is to save these to a separate config file and I'm sure if that's better? 🤷 I can just imagine that someone changes the filename, not realizing that the structure of the filename is encoding data that gets used elsewhere.

Reply:
yeah i had the same thought.

digging deeper into the past here, i think we wanted both:
- a way to go from the current account id that the developer is currently authenticated with (which we can get from aws sts get-caller-identity) to the tfbackend config file so that we can do `terraform init -backend-config=<current account's config>`
- we wanted the tfbackend config filename to be human readable e.g. dev.s3.tfbackend rather than <account id>.s3.tfbackend

so we decided to name the files `<human readable account name>.<account id>.s3.tfbackend`, then in our makefile we get the current backend config filename by searching for the current account id amongst the filenames (see [make infra-update-current-account](https://github.com/navapbc/template-infra/blob/90b7e18433eb21a3f46f24890c473b0849e27d49/Makefile#L90-L91) and [current-account-config-name.sh](https://github.com/navapbc/template-infra/blob/90b7e18433eb21a3f46f24890c473b0849e27d49/bin/current-account-config-name.sh))

given that we already have both the account name and account id in the filenames, i thought having a separate config file mapping account names to account ids would be less DRY.

an alternative would be to name the files `dev.s3.tfbackend`, without the account id, and have the account id in a separate file like `account_ids_by_name.json`, and then whenever we run `set-up-current-account` with a new account, the script would update the `account_ids_by_name.json` file. and to get the account name from the account id, we'd invert the map using jq like `cat account_ids_by_name.json | | jq 'to_entries | .[] | {(.value): .key}` and then use the resulting inverse map to get the account name by account id.

if the alternative is not much better, for now, i could also just add comments to all the places that rely on this filename structure:

- current-account-config-name.sh
- account-ids-by-name.sh
- .github/acounts/configure-aws-credentials/action.yml

thoughts?

Reply:
🤔 That's a tough one. I'm leaning towards the alternative that you described. I think that keeping the account IDs in a separate file is easier to approach.

Reply:
Mind if I push that off to a separate PR? It'll be a breaking change since I have to first copy over the <name>.<id>.s3.tfbackend files to <name>.tfbackend files and generate the separate config so that the scripts won't break when I update them, then update the scripts and finally remove the old files.

I created a ticket for it: https://github.com/navapbc/template-infra/issues/424

Reply:
Of course!